src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecuteWithErrorsTest.java

```diff
@@ -120,20 +119,209 @@ private void Repro47239Internal(String mode) throws Exception {
+            } catch (BatchUpdateException bue) {
+                actualUpdateCounts = bue.getUpdateCounts();
+                actualExceptionText = bue.getMessage();
+            }
+            for (int updateCount : actualUpdateCounts) {
+                log.fine("" + updateCount + ",");
+            }
+
+            assertTrue(Arrays.equals(actualUpdateCounts, expectedUpdateCounts),
+                    TestResource.getResource("R_errorFollow50280"));
+
+            // Test "soft" errors
+            conn.setAutoCommit(false);
+            stmt.addBatch(select);
+            stmt.addBatch(insertStmt);
+            stmt.addBatch(select);
+            stmt.addBatch(insertStmt);
+            try {
+                stmt.executeBatch();
+                // Soft error test: executeBatch unexpectedly succeeded
+                assertEquals(true, false, TestResource.getResource("R_shouldThrowException"));
+            } catch (BatchUpdateException bue) {
+                assertEquals("A result set was generated for update.", bue.getMessage(),
+                        TestResource.getResource("R_unexpectedExceptionContent"));
+                assertEquals(Arrays.equals(bue.getUpdateCounts(), new int[] {-3, 1, -3, 1}), true,
+                        TestResource.getResource("R_incorrectUpdateCount"));
+            }
+            conn.rollback();
+            stmt.addBatch(dateConversionError);
+            stmt.addBatch(insertStmt);
+            stmt.addBatch(insertStmt);
+            stmt.addBatch(insertStmt);
+            try {
+                stmt.executeBatch();
+            } catch (BatchUpdateException bue) {
+                if (isSqlAzureDW()) {
+                    assertThat(bue.getMessage(),
+                            containsString(TestResource.getResource("R_syntaxErrorDateConvertDW")));
+                } else {
+                    assertThat(bue.getMessage(), containsString(TestResource.getResource("R_syntaxErrorDateConvert")));
+                }
+            } catch (SQLException e) {
+                assertThat(e.getMessage(), containsString(TestResource.getResource("R_dateConvertError")));
+            }
 
+            conn.setAutoCommit(true);
+
+            // On SQL Azure, raising FATAL error by RAISERROR() is not supported and there is no way to
+            // cut the current connection by a statement inside a SQL batch.
+            // Details: Although one can simulate a fatal error (that cuts the connections) by dropping the
+            // database,
+            // this simulation cannot be written entirely in TSQL (because it needs a new connection),
+            // and thus it cannot be put into a TSQL batch and it is useless here.
+            // So we have to skip the last scenario of this test case, i.e. "Test Severe (connection-closing)
+            // errors"
+            // It is worthwhile to still execute the first 5 test scenarios of this test case, in order to have best
+            // test coverage.
```

Comment:
multi-line comments style (same in Repro47239largeInternal)

Reply:
Its the existing test+comments moved over to a different location hence shows changes like that. I don't think that's critical for this PR. Can be worked upon separately.

Reply:
it's not, I only brought this up cos I have another comment on this file :)